tests/drbg.cc

```diff
@@ -0,0 +1,362 @@
+/*
+ *    Copyright 2022 Cryspen Sarl
+ *
+ *    Licensed under the Apache License, Version 2.0 or MIT.
+ *    - http://www.apache.org/licenses/LICENSE-2.0
+ *    - http://opensource.org/licenses/MIT
+ */
+
+#include <fstream>
+#include <tuple>
+
+#include <gtest/gtest.h>
+#include <nlohmann/json.hpp>
+
+#include "EverCrypt_AutoConfig2.h"
+#include "Hacl_HMAC_DRBG.h"
+#include "Hacl_Spec.h"
+#include "hacl-cpu-features.h"
+
+#include "util.h"
+
+using namespace std;
+
+using json = nlohmann::json;
+
+typedef struct
+{
+  string hash;
+  vector<bytes> AdditionalInput;
+  bool PredictionResistance;
+  vector<bytes> EntropyInputPR;
+  uint32_t COUNT;
+  bytes EntropyInput;
+  bytes Nonce;
+  bytes PersonalizationString;
+  bytes EntropyInputReseed;
+  bytes AdditionalInputReseed;
+  bytes ReturnedBits;
+} CAVPTestCase;
+
+void
+cavp_to_hash(string value,
+             bool& skip_out,
+             Spec_Hash_Definitions_hash_alg& hash_out)
+{
+  skip_out = false;
+
+  if (value == "SHA-1") {
+    hash_out = Spec_Hash_Definitions_SHA1;
+  } else if (value == "SHA-224") {
+    skip_out = true;
+  } else if (value == "SHA-256") {
+    hash_out = Spec_Hash_Definitions_SHA2_256;
+  } else if (value == "SHA-384") {
+    hash_out = Spec_Hash_Definitions_SHA2_384;
+  } else if (value == "SHA-512") {
+    hash_out = Spec_Hash_Definitions_SHA2_512;
+  } else if (value == "SHA-512/224") {
+    skip_out = true;
+  } else if (value == "SHA-512/256") {
+    skip_out = true;
+  } else {
+    FAIL() << "Unexpected value \"" << value << "\".";
+  }
+}
+
+class DrbgSuite : public ::testing::Test
+{
+  void SetUp() override
+  {
+    hacl_init_cpu_features();
+    EverCrypt_AutoConfig2_init();
```

Comment:
You're not calling into evercrypt so this isn't necessary.
But feel free to keep it in as there will be evercrypt code coming up in #117

Reply:
👍🏻